lib/puppet/type/rabbitmq_user.rb

```diff
@@ -76,10 +76,4 @@ def should_to_s(newvalue = @should)
 
   end
 
-  validate do
-    if self[:ensure] == :present and ! self[:password]
```

Comment:
Removing this validation is not a valid solution to fixing self.instances. You can check both `self[:password].nil?` and `self.provider.password.nil?` eg: https://github.com/puppetlabs/puppetlabs-netapp/blob/master/lib/puppet/type/netapp_vserver.rb#L162